src/allotropy/parsers/biorad_bioplex/biorad_bioplex_structure.py

```diff
@@ -0,0 +1,272 @@
+                        if child.tag == MEMBER_WELLS:
+                            for member_well in child:
+                                sample_document = (
+                                    SampleDocumentAggregate._generate_sample_document(
+                                        sample_documents=sample_documents,
+                                        well_xml=member_well,
+                                        sample_id=sample_identifier,
+                                        sample_dilution=float(sample_dilution),
+                                        sample_type=sample_type,
+                                        sample_description=sample_description,
+                                    )
+                                )
+                                sample_documents.samples.append(sample_document)
+
+        return sample_documents
+
+    @staticmethod
+    def _generate_sample_document(
+        sample_documents: SampleDocumentAggregate,
+        well_xml: ElementTree.Element,
+        sample_id: str,
+        sample_dilution: float,
+        sample_type: str,
+        sample_description: str,
+    ) -> SampleDocument:
+        well_name = get_well_name(well_xml.attrib)
+        mappings = WellAnalyteMapping(well_name=well_name, analytes=[])
+        for analyte in well_xml[0]:
+            # Create the analyte.
+            new_analyte = AnalyteSample.create(analyte_xml=analyte)
+            # Add the analyte to the well analyte mappings.
+            mappings.analytes.append(new_analyte)
+            # Update the analyte region dict
+            sample_documents.analyte_region_dict[
+                new_analyte.analyte_region
+            ] = new_analyte.analyte_name
+        sample_document = SampleDocument(
+            sample_type=sample_type,
+            sample_identifier=sample_id,
+            description=sample_description,
+            well_name=well_name,
+            well_analyte_mapping=mappings,
+            sample_dilution=sample_dilution,
+        )
+        return sample_document
+
+
+@dataclass
+class DeviceWellSettings:
+    # This data class is for all metadata needed in the <Wells> section, used in measurement and device control docs.
+    well_name: str
+    sample_volume_setting: int
+    detector_gain_setting: str
+    minimum_assay_bead_count_setting: int
+    well_total_events: int
+    acquisition_time: str
+
+    @staticmethod
+    def create(well_xml: ElementTree.Element) -> DeviceWellSettings:
+        well_name = get_well_name(well_xml.attrib)
+        well_acq_time = get_val_from_xml(well_xml, "AcquisitionTime")
+        total_events = get_val_from_xml(well_xml, "TotalEvents")
+        sample_volume = int(get_val_from_xml(well_xml, "SampleVolume"))
+        # RP1 Gain. 16th element of Run Conditions
+        detector_gain_setting = get_val_from_xml(well_xml, "RP1Gain")
+        # Assuming second element of Run Settings
+        # Can't use util here because this pulls from attrib
```

Comment:
You could factor out the part that gets the element (e.g. `get_element_from_xml`) to get the element, then have `get_val_from_xml` call this to get the element, and then access the text.

Then, you can either make a `get_attribute_from_xml` that calls `get_element_from_xml` and then gets `.attrib[ATTRIBUTE_NAME]`. Or, if it's only used here, I'm find with you just calling the attrib part here.

The part I really want to avoid is hard-coded indices.